libraries/chain/controller.cpp

```diff
@@ -3882,7 +3882,7 @@ struct controller_impl {
    // and quorum_certificate_extension in block extension are valid.
    // Called from net-threads. It is thread safe as signed_block is never modified after creation.
    // -----------------------------------------------------------------------------
-   void verify_proper_block_exts( const block_id_type& id, const signed_block_ptr& b, const block_header_state& prev ) {
+   void verify_proper_block_exts( const std::optional<block_id_type>& id, const signed_block_ptr& b, const block_header_state& prev, bool complete_qc_validation ) {
```

Comment:
Maybe use `enum class full_qc_validation_t` instead of `bool`

Reply:
Sorry, now that I look at your change, I think it would be even cleaner to have:

`enum class qc_validation_t { no_bls_sig_validation, full };`

What do you think?

Reply:
We do two kinds of QC validation: basic (checking claimed block number, ...) and full (basic + BLS signature validation). Probably we should have `enum class qc_validation_t { basic, complete };` and add comments.

Reply:
Yes, but I thought it was nice if the `basic` name explicitely stated that we skipped the signature validation (that way no extra comments are needed).
What about: ` enum class qc_validation_t { basic_no_sig, complete };`?
If you prefer your simpler version that's fine too.